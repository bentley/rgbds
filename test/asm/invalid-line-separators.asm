; These directives expect `T_NEWLINE` specifically, not
; `endofline` which can be `T_NEWLINE | T_EOB | T_PERIOD`,
; since lexing IF/REPT/FOR/MACRO bodies uses real lines.

rept 3 . print "A1" . println "A2" . endr

rept 3 . print "B1" . println "B2"
endr

rept 3
	print "C1" . println "C2" . endr
	println "oops"
endr

for n, 1, \
       10 . println n . endr

if 1 . println "oops again"

macro foo
	println "foo" . endm
endm
	foo

macro bar . println "bar" . endm
	bar

macro baz . println "baz"
endm
	baz
