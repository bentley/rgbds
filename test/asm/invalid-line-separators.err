ERROR: invalid-line-separators.asm(5):
    syntax error, unexpected .
ERROR: invalid-line-separators.asm(5):
    syntax error, unexpected ENDR
ERROR: invalid-line-separators.asm(7):
    syntax error, unexpected .
ERROR: invalid-line-separators.asm(8):
    syntax error, unexpected ENDR
ERROR: invalid-line-separators.asm(10) -> invalid-line-separators.asm::REPT~1(11):
    syntax error, unexpected ENDR
ERROR: invalid-line-separators.asm(16):
    syntax error, unexpected .
ERROR: invalid-line-separators.asm(16):
    Expected constant expression: 'n' is not constant at assembly time
ERROR: invalid-line-separators.asm(16):
    syntax error, unexpected ENDR
ERROR: invalid-line-separators.asm(18):
    syntax error, unexpected .
ERROR: invalid-line-separators.asm(23) -> invalid-line-separators.asm::foo(21):
    syntax error, unexpected ENDM
ERROR: invalid-line-separators.asm(25):
    syntax error, unexpected ., expecting newline
ERROR: invalid-line-separators.asm(25):
    syntax error, unexpected ENDM
ERROR: invalid-line-separators.asm(26):
    Macro "bar" not defined
ERROR: invalid-line-separators.asm(28):
    syntax error, unexpected ., expecting newline
ERROR: invalid-line-separators.asm(29):
    syntax error, unexpected ENDM
ERROR: invalid-line-separators.asm(30):
    Macro "baz" not defined
error: Assembly aborted (16 errors)!
