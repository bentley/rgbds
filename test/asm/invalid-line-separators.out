A1A2
B1B2
C1C2
oops
$0
oops again
foo
bar
baz
